helpers/domain.cfc

```diff
@@ -0,0 +1,209 @@
+component accessors="true" {
+
+    property name="domain" default="";
+    property name="subdomain" default="";
+    property name="username" default="";
+    property name="ips" default="";
+    property name="custom_spf" default=false type="boolean";
+    property name="default" default=false type="boolean";
+    property name="automatic_security" default=false type="boolean";
+    property name="custom_dkim_selector" default="";
+  
+  
+    /**
+    * @hint Allow all values to be set by the init.   The domain name is the only required key needed to create an authenticated domain.
+    */
+    public any function init( required string domain, string subdomain, string username, array ips, boolean custom_spf, boolean default, boolean automatic_security, string custom_dkim_selector ) {
+
+        if ( arguments.keyExists( 'domain' ) )
+            this.domain( arguments.domain );
+
+        if ( arguments.keyExists( 'subdomain' ) OR arguments.subdomain NEQ "" )
```

Comment:
Just checking if there's a reason this (and a few other) checks are structured like this. It seems to me that `OR arguments.subdomain NEQ ""` is redundant. That is, if `arguments.subdomain NEQ ""`, then we already know that the `subdomain` key exists in `arguments`.

If that's the case, can you remove the `OR arguments.xxxx NEQ ""` portions of these checks?

If there's a reason for including them that I'm missing, just let me know.

Reply:
I believe these were supposed to be AND not OR conditions.   I wanted to exclude the property if an empty subdomain, username or custom_dkim_selector were passed in.   However, after taking another look I believe the propertyValues.filter should take care of this case so I will remove the NEQ conditions.